core/src/main/scala/flawless/syntax/syntax.scala

```diff
@@ -12,7 +12,22 @@ import cats.Show
 import cats.Eval
 import cats.Order
 
-object dsl {
+sealed trait NoDeps
+
+object NoDeps {
+  implicit val trivial: NoDeps = new NoDeps {}
+}
+
+final case class Location(file: String, line: Int)
+
+object Location {
+  implicit def fromSourceCode(implicit file: sourcecode.File, line: sourcecode.Line): Location = Location(file.value, line.value)
```

Comment:
The cool thing about this is that we could have zero dependencies on sourcecode (macros) whatsoever, and only provide a subtype of `anydsl` that's parameterized with Location in an optional artifact.

Now I only need to figure out how to embed this in the test values...

Reply:
thinking of having a type class for `CallsiteDeps` that will append appropriate information to the test output. The problem is, it needs to be able to plug into pretty much all the test's data (name, result etc.) and could attach information to any part of the output, so the output should probably be passed to the typeclass's function in a structured way.

Things that don't require information from the call site, like timing, can be passed to the interpreter instead.